staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/api.proto

```diff
@@ -1692,3 +1701,57 @@ enum ContainerEventType {
     // Container deleted
     CONTAINER_DELETED_EVENT = 3;
 }
+
+message ListMetricDescriptorsRequest {}
+
+message ListMetricDescriptorsResponse {
+    repeated MetricDescriptor descriptors = 1;
+}
+
+message MetricDescriptor {
+	// The name field will be used as a unique identifier of this MetricDescriptor,
+	// and be used in conjunction with the Metric structure to populate the full Metric.
+    string name = 1;
+    string help = 2;
+	// When a metric uses this metric descriptor, it should only define
+	// labels that have previously been declared in label_keys.
+	// It is the responsibility of the runtime to correctly keep sorted the keys and values.
+	// If the two slices have different length, the behavior is undefined.
+	repeated string label_keys = 3;
+}
+
+message ListPodSandboxMetricsRequest {} 
+
+message ListPodSandboxMetricsResponse {
+    repeated PodSandboxMetrics pod_metrics = 1;
+}
+
+message PodSandboxMetrics {
+    string pod_sandbox_id = 1;
+    repeated Metric metrics = 2;
+    repeated ContainerMetrics container_metrics = 3;
+}
+
+message ContainerMetrics {
+    string container_id = 1;
+    repeated Metric metrics = 2;
+}
+
+message Metric {
+    // Name must match a name previously returned in a MetricDescriptors call,
+    // otherwise, it will be ignored.
+    string name = 1;
+    // Return 0 if metric is cached
+    int64 timestamp = 2;
+    MetricType metric_type = 3;
+	// The corresponding LabelValues to the LabelKeys defined in the MetricDescriptor.
+	// It is the responsibility of the runtime to correctly keep sorted the keys and values.
+	// If the two slices have different length, the behavior is undefined.
+	repeated string label_values = 4;
+    UInt64Value value = 5;
+}
+
+enum MetricType {
+    COUNTER = 0;
+    GAUGE = 1;
```

Comment:
Do we foresee adding more types?

Reply:
the only other one the kubernetes metrics package supports is UNTYPED, so maybe? but it doesn't seem to be used in cadvisor rn and I'm not really clear on the usage